From a7f888ef39d15df6072aa41b2b4ead85426b2eb2 Mon Sep 17 00:00:00 2001
From: fujiwara <shintaro.fujiwara@gmail.com>
Date: Wed, 26 Aug 2020 05:21:42 +0900
Subject: [PATCH] fix sql only for postgres.

---
 classes/renderer.php | 11 ++++++++---
 version.php          |  4 ++--
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/classes/renderer.php b/classes/renderer.php
index 5c1a8b6..f6f7e14 100644
--- a/classes/renderer.php
+++ b/classes/renderer.php
@@ -182,10 +182,14 @@ class tool_managecourse_renderer extends plugin_renderer_base {
     }
 
     private function grade_sql($userid, $categoryid, $courseid) {
-
+        // for mysql
+        //$VIEW_COLUMNS="m.id as user_enrolments_id, k.name AS categoryname, u.id AS userid, u.firstname,
+        //               u.lastname, e.id enrolid, r.shortname AS rolename, c.fullname,
+        //               FROM_UNIXTIME(c.startdate) AS startdate, FROM_UNIXTIME(c.enddate) AS enddate,
+        //               g.finalgrade, g.rawgrademax";
         $VIEW_COLUMNS="m.id as user_enrolments_id, k.name AS categoryname, u.id AS userid, u.firstname,
                        u.lastname, e.id enrolid, r.shortname AS rolename, c.fullname,
-		       FROM_UNIXTIME(c.startdate) AS startdate, FROM_UNIXTIME(c.enddate) AS enddate,
+		       to_timestamp(c.startdate) AS startdate, to_timestamp(c.enddate) AS enddate,
                        g.finalgrade, g.rawgrademax";
         $FROM_TABLES="FROM {user} u, {user_enrolments} m, {enrol} e, {course} c, {role_assignments} a,
                       {role} r , {grade_items} i, {grade_grades} g, {course_categories} k";
@@ -212,7 +216,8 @@ class tool_managecourse_renderer extends plugin_renderer_base {
         if ($courseid) {
             $CONDITION3 = "AND c.id = $courseid";
         }
-        $GROUP_BY="GROUP BY m.id,u.id";
+        $GROUP_BY="GROUP BY m.id, u.id, e.id, c.id, k.name, r.shortname,
+                   c.fullname, c.startdate, c.enddate, g.finalgrade, g.rawgrademax";
         $ORDER="ORDER BY u.id, c.startdate, m.id, e.id, c.id";
         $DESC="DESC";
         $ASC="ASC";
diff --git a/version.php b/version.php
index e8ddb10..19ebe6a 100644
--- a/version.php
+++ b/version.php
@@ -25,7 +25,7 @@
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->component = 'tool_managecourse';
-$plugin->release = '0.1.46';
-$plugin->version = 2020082627;
+$plugin->release = '0.1.47';
+$plugin->version = 2020082628;
 $plugin->requires = 2018051700;
 $plugin->maturity = MATURITY_ALPHA;
-- 
2.26.2

